app/code/Magento/Email/Model/Template/Filter.php

```diff
@@ -1046,6 +1046,8 @@ public function applyInlineCss($html)
             try {
                 // Don't try to compile CSS that has compilation errors
                 if (strpos($cssToInline, ContentProcessorInterface::ERROR_MESSAGE_PREFIX)
+                    !== false ||
+                    strpos($cssToInline, 'undefined')
```

Comment:
@sivaschenko .. I have added this commit to fix the integration test for https://github.com/magento/magento2/blob/2b21388fcf8d8cf1d03220cd8893b88aeb6f9efc/dev/tests/integration/testsuite/Magento/Email/Model/Template/FilterTest.php#L423
Can you please review this? I feel we need to handle this in different way
Thanks

Reply:
@konarshankar07 Looks like emogrifier is no longer actually working with CSS string containing the error message.

I think the best way to handle the CSS parsing error is to throw an exception in `\Magento\Email\Model\Template\Filter::getCssFilesContent` and handle it in client code.
Instead of returning an exception message as the result css and then parsing the result for error messages.

Thank you!